### DISCUSSION TITLE
Is there any way to easily remove menu bar items?

### DISCUSSION THREAD

**Discussion by mamrosh** [GMOD/jbrowse-components#4191] [General]
Hi,

My group is going to use MsaView in the near future on our website. However, we would like to remove many of the menu bar items (e.g. "Bookmarks" under "Tools" in the top black menu bar, or 'Add track' in the MsaView-specific menu bar), for a few reasons: (1) some of them aren't compatible with MsaView (2) we don't want to confuse our users by giving them options to view their own alignments on our website, especially since we have modified the MsaView code to include custom residue numbering and a few other things. There seems to be an easy way to add menu items but not to remove them. I could go into the code and try to remove  each one we don't want individually, but I'm hoping there's an easier way! Thanks!

**Comment by cmdcolin** [comment]
note that if you are only interested in using the MSA view, then you can embed that on its own without any jbrowse at all. an example of this is at https://gmod.github.io/react-msaview/


there are programmatic ways of manipulating the top menu bar of jbrowse 2, it is a little bit involved and a better interface could probably be made for this. the setMenus function particularly can be used to remove things. https://jbrowse.org/jb2/docs/models/rootappmenumixin/ you would have to make a custom plugin to use that. If interested, I can elaborate

**Comment by mamrosh** [comment]
Thanks for this fast response. If you could elaborate a little, that would be helpful. I like how MsaView looks when embedded with jbrowse, so I would like to keep that if possible. Is it possible to write a custom plugin to remove jbrowse menu items and run it from msa_plugin.js ? I wonder if the easiest solution would be to remove all the menus and then I can re-add the ones that I want (which are very few if any of what is currently under "File", "Add", etc.). I can use the "appendToMenu" and similar to set up a custom menu. I have been trying to figure out how to make it so that setMenus results in no menus (and I later add the ones I want) but I haven't been successful yet. Sorry, I have been able to figure out how to modify msa_plugin.js to my needs (and really like MsaView overall!) but the jbrowse2 code is much more complicated.

**Reply by cmdcolin** [reply]
@mamrosh you can add this line to your plugin's "configure" method

```typescript
class Plugin {
  install(pluginManager) {
    //do stuff...
  }
  configure(pluginManager) {
    //run this in the configure method. note that the rootModel might not be available in some contexts like web worker so guarded in if statement
    if(pluginManager.rootModel) {
      pluginManager.rootModel.setMenus([])
    }
  }
}
```

result: no menus :)

![image](https://github.com/GMOD/jbrowse-components/assets/6511937/8bee3adc-637b-478d-8f67-a96c4fe7ec90)

this is a random thing I will also note but I am doing a lot of development work on the MSA view plugin currently. if you are interested let me know. it is developing features like mouse-over mapping between the linear genome view and the positions of a protein in the MSA view

**Reply by cmdcolin** [reply]
let me know also if you have any particular MSA view features you are working on too...the coordinate things might be something I could add to the core instance also
